pkg/policy/api/icmp.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import "strconv"
+
+const (
+	IPv4Family = "IPv4"
+	IPv6Family = "IPv6"
+)
+
+type ICMPRules []ICMPRule
+
+// ICMPRule is a list of ICMP fields.
+type ICMPRule struct {
+	// Fields is a list of ICMP fields.
+	//
+	// +kubebuilder:validation:Optional
+	Fields []ICMPField `json:"fields,omitempty"`
+}
+
+// ICMPField is a ICMP field.
+type ICMPField struct {
+	// Family is a IP address version.
+	// Currently, we support `IPv4` and `IPv6`.
+	// `IPv4` is set as default.
+	//
+	// +kubebuilder:default=`IPv4`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern=`IPv(4|6)`
```

Comment:
While checking it against the pattern will work, we can use the `Enum` validation because we only expect two values:

```suggestion
	// +kubebuilder:validation:Enum=IPv4;IPv6
```